content/en/docs/concepts/extend-kubernetes/compute-storage-net/resource-drivers.md

```diff
@@ -0,0 +1,286 @@
+initialization and setup.
+
+With device plugins, the scheduler was given a trivial, numerical representation of the resources
+available on node for consideration during scheduling as an extended resource.
+
+With DRA, the scheduler is offloading the task of allocating and accounting for non-native resources
+to the resource driver, which manages such resources in the cluster.
+
+Resource driver consists of two main components:
+
+- a _controller_ (one per cluster), manages hardware resources allocation for
+  {{< glossary_tooltip term_id="resource-claim" text="ResourceClaims">}}
+- _kubelet-plugin_ (one per node with resource), that:
+  - discovers the supported hardware
+  - announces the discovered hardware to the Resource Driver controller
+  - prepares the hardware allocated for Pod when Pod comes to the Kubelet
+  - unprepares the hardware allocated for a Pod when the Pod has reached final state or is being deleted.
+
+Communication between the controller and the kubelet-plugin can be realized in two ways:
+
+- through {{< glossary_tooltip term_id="CustomResourceDefinition" text="CRD">}} objects provided by
+  and designed for the resource driver
+- through a ResourceHandle which is a part of an `AllocationResult` provided by the controller in case of
+  successful allocation
+
+General recommendations:
+
+- resource driver name pattern: `<HW type>.resource.<companyname>.<companydomain>`. For example,
+  gpu.resource.example.com
+
+<!-- body -->
+
+## Resource driver registration
+
+Controller and kubelet-plugin registration can be handled with the DRA _helper code_ that implements
+common for all resource drivers functionality.
+
+### Controller registration
+
+Resource driver controller has to implement a [driver interface](https://pkg.go.dev/k8s.io/dynamic-resource-allocation@v0.28.1/controller#Driver),
+which then can be simply used with [controller helper instance](https://pkg.go.dev/k8s.io/dynamic-resource-allocation@v0.28.1/controller#New).
+
+See [example resource driver controller](https://github.com/kubernetes-sigs/dra-example-driver/blob/151b7c8da2e620c47a3591e1a937f8d0297b0c25/cmd/dra-example-controller/main.go#L208) for details
+
+
+### Kubelet-plugin registration
+
+Resource driver kubelet-plugin has to implement [NodeServer interface](https://pkg.go.dev/k8s.io/kubelet@v0.28.2/pkg/apis/dra/v1alpha3#NodeServer),
+which can be simply used with [kubelet-plugin helper instance](https://github.com/kubernetes-sigs/dra-example-driver/blob/151b7c8da2e620c47a3591e1a937f8d0297b0c25/cmd/dra-example-kubeletplugin/main.go#L160).
+
+## Controller implementation
+
+Controller's main responsibility is to allocate and deallocate resource for
+{{< glossary_tooltip term_id="resource-claim" text="ResourceClaim">}}.
+
+There are two modes of allocation the ResourceClaim can have:
+
+- `WaitForFirstConsumer` (default), AKA "delayed". In this mode the resource will only be requested
```

Comment:
```suggestion
- `WaitForFirstConsumer` (default), which you could think of as meaning _delayed_.
  In this mode the cluster only requests resource(s)
```

Reply:
```suggestion
- `WaitForFirstConsumer` (default), also known as _delayed allocation_. In this mode the resources are only requested
```

It literally says Wait in the beginning of the name, so it's hard to not think about it as not delayed IMHO. The `AKA "delayed"` meant to signify that many places (slack, KEP, PRs, community meetings) this mode _is_ called delayed, nobody uses the full name of `WaitForFirstConsumer`, "delayed" is a commonly used name for it, not just understanding how it works.

Resource driver is a part of cluster, so the latter sentence should be something else than "... cluster only requeste ...", perhaps `In this mode the resources are only requested` ?